### PR TITLE
Serve old financials when new financials are missing

### PR DESCRIPTION
**Pull request by ValueRaider**
For some tickers Yahoo does not display quarterly financials. I notice most with Australian tickers e.g. BSE.AX . https://finance.yahoo.com/quote/BSE.AX/cash-flow?p=BSE.AX

This means `yfinance` using the TimeSeries API also returns an empty table. But the old-style table is still present in QuoteTimeSeriesStore, so I propose returning this. Question - should user be alerted to switch? How?

@git-shogg

### CODE REVIEW

**Comment by git-shogg** [comment]
@ValueRaider good point. I am fine either way the main thing to note here is that TTM (Trailing Twelve Months) is not a quarterly type metric at all.

**Comment by fredrik-corneliusson** [comment]
Do you have an example of how to test this?
I tried to test this by adding one to TestTickerMiscFinancials but it fails with exception.
```
    def test_quarterly_income_statement_old_fallback(self):
        expected_row = "TotalRevenue"
        ticker = yf.Ticker("BSE.AX")
        data = ticker.quarterly_income_stmt
        self.assertIsInstance(data, pd.DataFrame, "data has wrong type")
        self.assertFalse(data.empty, "data is empty")
        self.assertIn(expected_row, data.index, "Did not find expected row in index")

        data_cached = self.ticker.quarterly_income_stmt
        self.assertIs(data, data_cached, "data not cached")

```

**Comment by fredrik-corneliusson** [comment]
> I propose returning this. Question - should user be alerted to switch? How?
Do you mean user should get a message informing on no data could be fetched with the new method and that they could try the old using a separate function? If that is the case I would very much advocate for that approach.

On a more general note.
A lot of effort has been put into cleaning up the code in order to make the code less bug prone, faster and easier on the yahoo requests. I'm afraid that if we begin adding quite high level if:s and but:s deep in the API it will quickly negate that effort and harm the project in the long run. 
I would suggest having a core API that is simple and flexible and if needed add a layers on top of it to create more specialized higher level APIs.

Just my 2 cents.

What are your thoughts?

**Comment by ValueRaider** [comment]
I'm fine with the general note, just unsure about default behaviour. @git-shogg do you prefer automatic falling-back to old-format financials data with a message? Or should user have to specifically request via an argument?

**Comment by git-shogg** [comment]
I think it's fine to revert back to the old format. You may have an issue with the camelcase function reading things that are in all caps (e.g. "EBITDA")? Is there an issue here or does it handle that ? Can always add the optional argument later.

**Comment by fredrik-corneliusson** [comment]
Well automatic reverting back to old format will require much bigger change that current PR

**Comment by ValueRaider** [comment]
Auto-reverting is actually simple. Add more getters to the fundamentals.py, and switching logic to base.py. I'll push soon.

**Comment by ValueRaider** [comment]
Auto-reverting now improved - logic in `Ticker`, fetching refactored in `fundamentals.py`

**Review by fredrik-corneliusson — CHANGES_REQUESTED** [review]

File: yfinance/scrapers/fundamentals.py [hunk 1]
```diff
@@ -132,6 +133,13 @@ def _scrape(self, name, timescale, proxy=None):
 
         try:
             statement = self._create_financials_table(name, timescale, proxy)
+
+            if statement.shape[0] == 0:
+                # Normally table only empty when nothing on Yahoo. So good?
```
[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/scrapers/fundamentals.py]: It's meant as a rhetorical question - "If Yahoo not displaying a table anyway, is there actually a problem?"

File: yfinance/scrapers/fundamentals.py [hunk 2]
```diff
@@ -132,6 +133,13 @@ def _scrape(self, name, timescale, proxy=None):
...
+                # Except 'QuoteSummaryStore' still contains the old financial data,
```
[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/scrapers/fundamentals.py]: Old way of fetching

File: yfinance/scrapers/fundamentals.py [hunk 3]
```diff
@@ -155,6 +163,34 @@ def _create_financials_table(self, name, timescale, proxy):
         except Exception as e:
             pass
 
+    def _create_financials_table_old(self, name, timescale, proxy):
```
[the remark this answers is not in the document]
Comment (ValueRaider) [reply yfinance/scrapers/fundamentals.py]: Same columns, but different index and values. Mystery why Yahoo maintains two different tables for each financials.

**Comment by ValueRaider** [comment]
@git-shogg Any opinion on this? Ideally can you try this branch, with a ticker that requires falling-back e.g. BSE.AX, then provide any feedback?

**Comment by ValueRaider** [comment]
Removed the falling back, replaced with toggle argument defaulting to new format.

**Pull request merged**